tools/js-sdk-release-tools/packages/typescript-codegen-breaking-change-detector/src/tests/patch-detection.test.ts

```diff
@@ -1,183 +1,333 @@
+    expect(breakingPairs[0].location).toBe(DiffLocation.Signature_ParameterList);
+    expect(breakingPairs[0].target?.name).toBe('funcParameterCount');
 
-      breakingPairs = patchFunction('funcRemove', astContext);
-      expect(breakingPairs.find((p) => p.assignDirection !== AssignDirection.CurrentToBaseline)).toBeUndefined();
-      expect(breakingPairs.length).toBe(1);
-      expect(breakingPairs[0].reasons).toBe(DiffReasons.Removed);
-      expect(breakingPairs[0].location).toBe(DiffLocation.Signature);
-      expect(breakingPairs[0].target?.name).toBe('funcRemove');
+    breakingPairs = patchFunction('funcParameterType', astContext);
+    expect(breakingPairs.find((p) => p.assignDirection !== AssignDirection.CurrentToBaseline)).toBeUndefined();
+    expect(breakingPairs.length).toBe(1);
+    expect(breakingPairs[0].reasons).toBe(DiffReasons.TypeChanged);
+    expect(breakingPairs[0].location).toBe(DiffLocation.Parameter);
+    expect(breakingPairs[0].target?.name).toBe('a');
 
-      breakingPairs = patchFunction('funcAdd', astContext);
-      expect(breakingPairs.find((p) => p.assignDirection !== AssignDirection.CurrentToBaseline)).toBeUndefined();
-      expect(breakingPairs.length).toBe(1);
-      expect(breakingPairs[0].reasons).toBe(DiffReasons.Added);
-      expect(breakingPairs[0].location).toBe(DiffLocation.Signature);
-      expect(breakingPairs[0].source?.name).toBe('funcAdd');
-    } finally {
-      if (tempFolder) remove(tempFolder);
-    }
+    breakingPairs = patchFunction('funcRemove', astContext);
+    expect(breakingPairs.find((p) => p.assignDirection !== AssignDirection.CurrentToBaseline)).toBeUndefined();
+    expect(breakingPairs.length).toBe(1);
+    expect(breakingPairs[0].reasons).toBe(DiffReasons.Removed);
+    expect(breakingPairs[0].location).toBe(DiffLocation.Signature);
+    expect(breakingPairs[0].target?.name).toBe('funcRemove');
+
+    breakingPairs = patchFunction('funcAdd', astContext);
+    expect(breakingPairs.find((p) => p.assignDirection !== AssignDirection.CurrentToBaseline)).toBeUndefined();
+    expect(breakingPairs.length).toBe(1);
+    expect(breakingPairs[0].reasons).toBe(DiffReasons.Added);
+    expect(breakingPairs[0].location).toBe(DiffLocation.Signature);
+    expect(breakingPairs[0].source?.name).toBe('funcAdd');
   });
 
+  // TODO: seperate tests
   test('detect routes', async () => {
-    const currentApiViewPath = join(__dirname, testCaseDir, 'current-package/patch.api.md');
-    const baselineApiViewPath = join(__dirname, testCaseDir, 'baseline-package/patch.api.md');
-    const date = getFormattedDate();
-
-    let tempFolder: string | undefined = undefined;
-    try {
-      const tempFolder = await createTempFolder(`.tmp/temp-${date}`);
-      const astContext = await createAstContext(baselineApiViewPath, currentApiViewPath, tempFolder);
-      const breakingPairs = patchRoutes(astContext);
-      expect(breakingPairs.length).toBe(5);
+    const baselineApiView = `
```

Comment:
Could you help me understand what the changelog would look like for routes change?

Reply:
this pr is [Part 1], doesn't have impact in change log tool, so changes in changelog. next PR will update the changelog, I can show you there